### PR TITLE
Updating deps

### PR DESCRIPTION
**Pull request by chrisnojima**
This is a branch off of https://github.com/keybase/client/pull/3351 .  We need a new dep (null-loader) which causes us to have to remake the shrinkwrap. Material-ui pulls in newer react which causes a chain reaction of peer dep problems. This PR updates (basically everything). Nothing seemed to really change much except menubar which just changed some options camelCasing. Since i was there i put in a little bit of code to make double clicking better (its still problematic but feels better than what we currently have).

@keybase/react-hackers

### CODE REVIEW

File: desktop/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "postinstall": "babel-node --presets es2015,stage-2 npm-helper.js postinstall",
     "render-screenshots": "babel-node --presets es2015,stage-2 npm-helper.js render-screenshots",
     "updated-fonts": "babel-node --presets es2015,stage-2 npm-helper.js updated-fonts",
+    "wrap": "rm -rf node_modules/msgpack && npm shrinkwrap --dev",
```
Comment (chrisnojima) [inline desktop/package.json]: lets just make a shortcut for this

File: shared/login/login/index.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ class Login extends Component {
   }
 
   render () {
-    return <Render { ...this.props }
+    return <Render {...this.props}
```
Comment (chrisnojima) [inline shared/login/login/index.js]: new eslint catches this

**Comment by chrisnojima** [comment]
- [ ] wait till visdiff fix is in

File: desktop/package.json (hunk 1 above)
Comment (chromakode) [inline desktop/package.json]: Would `npm prune` work as an alternative to the `rm`? That way if you have any other extra packages hanging around they get cleaned up too.

**Comment by chromakode** [comment]
:+1:

**Comment by chrisnojima** [comment]
will merged down then rebase to get visdiff fixes

**Pull request merged**